lib/SampleService/core/validator/builtin.py

```diff
@@ -314,3 +317,52 @@ def _is_num(name, val):
     if val is not None and type(val) != float and type(val) != int:
         raise ValueError(f'Value for {name} parameter is not a number')
     return val
+
+def ontology_has_ancestor(d: Dict[str, Any]) -> Callable[[str, Dict[str, PrimitiveType]], Optional[str]]:
+    '''
+    Build a validation callable that checks that value has valid ontology ancestor term provided
+
+    The 'ontology' parameter is required and must be a string. It is the ontology name.
+
+    The 'ancestor_term' parameter is required and must be a string. It is the ancestor name.
+
+    :param d: the configuration map for the callable.
+    :returns: a callable that validates metadata maps.
+    '''
+    _check_unknown_keys(d, {'ontology', 'ancestor_term'})
+
+    srv_wiz_url=os.environ.get('KBASE_ENDPOINT', 'https://kbase.us/services/service_wizard').strip('/') + '/service_wizard'
+
+    ontology = d.get('ontology')
+    if not ontology:
+        raise ValueError('ontology is a required paramter')
+    if type(ontology) != str:
+        raise ValueError('ontology must be a string')
+
+    ancestor_term = d.get('ancestor_term')
+    if not ancestor_term:
+        raise ValueError('ancestor_term is a required paramter')
+    if type(ancestor_term) != str:
+        raise ValueError('ancestor_term must be a string')
+
+    def _get_ontology_ancestors(ontology, val):
+        try:
+            oac=OntologyAPI(srv_wiz_url)
+            ret=oac.get_ancestors({"id": val, "ns": ontology})
+            return list(map(lambda x: x["term"]["id"], ret["results"]))
+        except ServerError as err:
+            if 'InvalidParamsError' in err.data:
+                raise ValueError(f'ontology {ontology} is not found')
```

Comment:
This still seems odd to me - if the service is passing invalid parameters, why are we reporting an error that the ontology isn't found? Surely that's a different error?